com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -458,13 +458,16 @@ UnityRLInitializationOutputProto GetTempUnityRlInitializationOutput()
             {
                 if (m_CurrentUnityRlOutput.AgentInfos.ContainsKey(behaviorName))
                 {
-                    if (output == null)
+                    if (m_CurrentUnityRlOutput.AgentInfos[behaviorName].CalculateSize() > 0)
```

Comment:
Can you explain why this is necessary? How could the AgentInfos for a behavior name be in the dictionary but the list be empty?

Reply:
I am still investigating that part.

Reply:
(discussed on slack) This change sounds OK, but can you add some comments to clarify what's going on (my bad for not documenting better in the first place). Something like "Check the BrainParameters that we haven't sent yet, and add them to the InitializationOutput if we have observations for that behavior this step."